### DISCUSSION TITLE
HTTPS / External storage support

### DISCUSSION THREAD

**Discussion by needsupport** [bitmagnet-io/bitmagnet#183] [Ideas]
can we add HTTPS support at priority please since?

Also, Postgres currently is not working with external storage that is mounted on ubuntu. can we do that.

Thanks

**Comment by DyonR** [comment]
Just out of curiosity; why do you find HTTPS such an high priority?

**Comment by mgdigital** [comment]
HTTPS might best be done via a reverse proxy. I don't know that it's something Bitmagnet would have to specifically implement but if someone has a nice example using something like Traefik or nginx I'd be happy to add it to the docs.

> Also, Postgres currently is not working with external storage that is mounted on ubuntu. can we do that.

I think this can be done with Postgres tablespaces. Again I don't have a nice example and am happy to add to the docs if someone else does.

**Comment by needsupport** [comment]
Given that this might scope up all sort of information on the internet, I want to be careful about how this runs. Apart from running this behind a VPN, I am running this on a VPS that has no open ports at all. This is why I don't want to run it behind a reverse proxy. So I use cloud flare tunnels and HTTPS would be very helpful in keeping this information private.

About the Postgres problem, I am adding about 400K torrents in about 24 hours. So I can not run this on a cheap VPS and have enough space. I can however mount an external cifs/SAMBA .


I think the requests that I have made here are to ensure that in the remote possibility that bitmagnet hoovers up questionable stuff on the internet, the data I always shared over a secure HTTPS tunnel and stored in a different place than where the VPS is.
happy to share logs/ more info to help here.

**Comment by davispuh** [comment]
If you are exposing it to public internet then just put it behind reverse proxy (eg. Nginx). That works perfectly fine. Most users aren't running just `bitmagnet` alone but other services aswell so it makes sense to put everything behind Nginx. I think terminating HTTPS at application layer is waste of time. For example I want all the features HTTP/3, TLS 1.3, Ed25519 which is unlikely to be supported by applications and huge effort for them.

Now if you aren't exposing it to public but access it using backchannels such as over SSH then you don't need HTTPS because whatever underlying channel you use it already would be encrypted by that protocol so you don't gain anything anyway.

> About the Postgres problem, I am adding about 400K torrents in about 24 hours. So I can not run this on a cheap VPS and have enough space. I can however mount an external cifs/SAMBA .

I haven't tried this but I don't see issue with this, it should work. Just setup [PostgreSQL tablespaces](https://www.postgresql.org/docs/current/manage-ag-tablespaces.html). But I recommend trying SSHFS. I'm using tablespace on local filesystem and that works without any issues so I don't think there would be any issues with remote one aswell.
